schemas/json/block.json

```diff
@@ -256,7 +256,22 @@
 				},
 				"className": {
 					"description": "By default, the class .wp-block-your-block-name is added to the root element of your saved markup. This helps having a consistent mechanism for styling blocks that themes and plugins can rely on. If, for whatever reason, a class is not desired on the markup, this functionality can be disabled.",
-					"type": "boolean",
+					"oneOf": [
+						{ "type": "boolean" },
+						{
+							"type": "object",
+							"properties": {
+								"block": {
+									"description": "Add the .wp-block-your-block-name class to the block's wrapper element",
+									"type": "boolean"
+								},
+								"variation": {
+									"description": "Add the .wp-block-your-block-name-your-block-variation class to the block's wrapper element",
```

Comment:
Should it get clarified that this class name gets injected only when the active variations can be detected during saving?

Reply:
Hmm, but isn't that equivalent to active variation detection working at all on the client side? I.e. if a block is correctly shown e.g. in the block inspector as a variation, then the class name will get injected correctly as well. (As opposed to the server side, where active variation detection is a bit trickier.)

If it's equivalent to "the active variation is detected correctly [on the client]", then IMO we don't need to point that out 🤔